spec/libs/installations/crashlytics/api_spec.rb

```diff
@@ -0,0 +1,189 @@
+require "rails_helper"
+
+Dataset = Struct.new(:dataset_id, :project_id)
+APP_IDENTIFIER = "com.example.app"
+APP_ONE = "App One"
+APP_TWO = "App Two"
+
+describe Installations::Crashlytics::Api, type: :integration do
+  let(:project_number) { Faker::Number.number(digits: 8).to_s }
+  let(:json_key) { StringIO.new({client_email: "client@test.com", private_key: "private_key"}.to_json) }
+  let(:api_instance) { described_class.new(project_number, json_key) }
+
+  let(:app_id) { APP_IDENTIFIER }
+  let(:version) { "1.0.0" }
+  let(:build_number) { "100" }
+  let(:payload) do
+    {
+      new_errors_count: 5,
+      errors_count: 10,
+      sessions_in_last_day: 100,
+      sessions: 200,
+      sessions_with_errors: 50,
+      daily_users_with_errors: 30,
+      daily_users: 150,
+      total_sessions_in_last_day: 150,
+      external_release_id: "release_123"
+    }
+  end
+  let(:transforms) { CrashlyticsIntegration::RELEASE_TRANSFORMATIONS }
+
+  describe "#find_release" do
+    let(:app_id) { Faker::Alphanumeric.alphanumeric(number: 8) }
+    let(:app_version) { "1.0.0" }
+    let(:app_version_code) { "100" }
+    let(:bundle_identifier) { APP_IDENTIFIER }
+    let(:transforms) { CrashlyticsIntegration::RELEASE_TRANSFORMATIONS }
+    let(:payload) do
+      {
+        version_name: app_version,
+        errors_count: 10,
+        new_errors_count: 5,
+        sessions: 200,
+        sessions_in_last_day: 100,
+        sessions_with_errors: 50,
+        daily_users: 150,
+        daily_users_with_errors: 30,
+        external_release_id: "release_123",
+        total_sessions_in_last_day: 150
+      }
+    end
+
+    context "when release data is found" do
+      it "returns the transformed release data" do
+        allow(api_instance).to receive(:fetch_crash_data)
+          .with(app_id, app_version, app_version_code, bundle_identifier)
+          .and_return(payload)
+
+        result = api_instance.find_release(app_id, app_version, app_version_code, transforms, bundle_identifier)
+        expected_data = {
+          "daily_users" => 150,
+          "daily_users_with_errors" => 30,
+          "errors_count" => 10,
+          "external_release_id" => "release_123",
+          "new_errors_count" => 5,
+          "sessions" => 200,
+          "sessions_in_last_day" => 100,
+          "sessions_with_errors" => 50,
+          "total_sessions_in_last_day" => 150
+        }
+
+        expect(result).to eq(expected_data)
+      end
+    end
+
+    context "when no release data is found" do
+      it "returns nil" do
+        allow(api_instance).to receive(:fetch_crash_data)
+          .with(app_id, app_version, app_version_code, bundle_identifier)
+          .and_return(nil)
+
+        result = api_instance.find_release(app_id, app_version, app_version_code, transforms, bundle_identifier)
+
+        expect(result).to be_nil
+      end
+    end
+  end
+
+  it "#list_apps" do
```

Comment:
**todo:** Because of the new changes, this test can be moved to the firebase api (I don't think that layer has any tests) since this is a specifically firebase api spec.